### PR TITLE
Hyperparameter Tuning In Decision Trees and Random Forests

### LINKED ISSUE #5569: [Machine Learning] Hyperparameter Tuning in Decision Trees and Random Forests with Python
### Topic Suggestion
[Machine Learning] Hyperparameter Tuning in Decision Trees and Random Forests using Python

#### Proposed title of the article
Hyperparameter Tuning in Decision Trees and Random Forests

#### Proposed article introduction
[Machine Learning] Introduction to ML

After years of hard work, we've reached a stage where we are using computers to analyze millions of data points and provide insights that even the human eye couldn't catch. But our machine Learning Model is only as good as its accuracy on unseen data, i.e how well our model generalizes. In this article, I'll solve a Binary Classification Problem, using a Decision Tree Classifier and Random forests and solve the overfitting problem by tuning their Hyperparameter and compare results. 
This shall be a beginner-friendly article for those new to Decision Trees.  

#### Key takeaways 
- Introduction to ML
- Elaborated Introduction to  Decision Trees and Random Forests
- Gini Index and Feature Importance
- Overfitting and Regularisation 
- Ensembling
- Tuning Hyperparameters: `max_depth`, `max_leaf_nodes`, `n_estimators` etc
 

#### Article quality
This is a unique article from the published articles on the section.io/engineering-education. Understanding the concept of overfitting and learning how to overcome it is as important as preprocessing the data and making it fir for modeling. 
Similarly, tuning the decision tree hyperparameters is as important as learning how to use this approach in ML, which will also be covered in-depth in this article.

### PR DESCRIPTION
**Pull request by himani007**
## Our **Peer Reviewer and Content Moderator teams do NOT provide any revisions services.** All revisions and edits should be completed on your own forked repo (as to not take up room in the queue) in order for our team to review them all in a timely manner. 

**ANY ARTICLE SUBMITTED WITH GLARING ERRORS WILL BE IMMEDIATELY CLOSED.**

>As a rule of thumb - please be sure to only submit articles (pull requests) that are fully polished and ready to be published. Be sure to go through our resources documents for extra and 3rd party (vetted) resources to help improve overall technical writing.

> If you have not already, please go over our [Resources Page](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/resources-page.md) for more tips and tricks on how to improve your overall technical writing so reviews are swifter, increase the chance of a payout and provide more value to our readers.

> For first-time contributors and for more details on our submission guidelines, **see our [Contributing Guidelines](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/CONTRIBUTING.md)**.

#### Formatting and structure checks
- [x] Have you used the correct folder and file structure? - **see [Contribution Guidelines](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/CONTRIBUTING.md)**
- [x] Is your article properly formatted in Markdown? - **see [Github Markdown guide](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet)**
- [x] Have you used our preferred heading structure? - **[H3 (###) and up](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet#headers)**
- [x] Does your article follow any of the suggested structure formats? - **see [example formats](https://github.com/section-io/engineering-education/blob/master/CONTRIBUTING.md#example-format-structure)**

- [x] Is your article over 750 words? - **Tool**: [Wordcounter.net](https://wordcounter.net)
- [x] Is the article you are submitting an in-depth and unique article? Does it go beyond what is in the official docs and what is covered in other blog sites? See these articles as [examples](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/resources-page.md).
- [x] Does your article provide enough value and detail about your topic? - **Articles should be clear, accurate, and fully explained.**
- [x] Can your article be understood by beginners? - **Assume the audience is smart but has no prior exposure to the common terminology in your article.**

- [x] Have you included a hero image that is **600x400 pixels**, under **300KB in size**, and have the copyright to use it? - **Tools:** [Pixlr Image Editor](https://pixlr.com/e) for resizing and [Unsplash](https://unsplash.com) for Creative Commons images
      
#### Grammar & plagiarism checks
- [x] Have you spellchecked and grammar checked your article? - **Tools:** [Write&Improve](https://writeandimprove.com/?lang=en-GB) - [ProWritingAid](https://prowritingaid.com/) - [Grammarly](https://grammarly.com) 
- [x] Please place your article through a 3rd party plagiarism checker. We suggest using Quetext, [this tool](https://www.quetext.com/) is free and has a daily limit. **We typically accept articles with 10% or less.**
- [x] Have you checked your article for readability? - **Tool:** [Hemmingway](http://www.hemingwayapp.com)
- [x] Have you added sources for quotes and images that aren't yours?

#### Technical checks
- [x] Are your code snippets properly formatted for syntax highlighting - **see [Syntax guide](https://gohugo.io/content-management/syntax-highlighting)**
- [x] Have you checked your code runs correctly and you've highlighted all necessary dependencies for installation?
- [x] Are the software programs and packages you're highlighting in your article up to date, using current versions and not deprecated?

This closes #5569 

> Finally, delete the article checklist notes in blockquotes and submit your PR. We look forward to reviewing your article.

### CODE REVIEW

**Comment by srishilesh** [comment]
@himani007 
Welcome to EngEd community!

Your PR has entered our reviewing queue. Please wait until a peer-reviewer pings you to get started with the review process.

To avoid potential closure of the PR:
- Perform a grammar check
- Make sure the hero image is of dimension 600x400
- Start markdown headers from `###`
- Go through the markdown guidelines that we follow here, by referring to our contribution guidelines
- For using bulleted points, use - instead of *
- For code snippets, make sure to explain them properly or add comments wherever needed
- The image paths are correct
- The author info is correct (Refer to the guidelines to view the latest update in author info section)
- The directory names are in lower case, separated by `-`

We also suggest you to go through other approved PRs to see how the authors have formatted their articles.

Thanks.

**Comment by ahmadmardeni1** [comment]
Plagiarism check done ✅

**Comment by ahmadmardeni1** [comment]
Upon running your article through our 3rd party plagiarism checker, it seemed to raise a few flags and the % was higher than we typically accept. @himani007 
Please see attached PDF - and revisit the article to ensure we are contributing wholly unique and original content.
Be sure to see our resources page to see more info on plagiarism and what is considered as such.

[Article on Hyperparameter tuning#6066](https://github.com/section-engineering-education/engineering-education/files/7890603/New.Scan.4_10.PM.Copyleaks.Report.pdf)

**Comment by himani007** [comment]
Hi Ahmad, 
I have revised the article into minimum plagiarism as per the required policies and committed the changes. 
Please have a look and let me know if the changes have been made..

**Comment by ahmadmardeni1** [comment]
The % is still high. Please fix the plagiarism. @himani007
cc @Bayler 
[Article on Hyperparameter tuning#6066](https://github.com/section-engineering-education/engineering-education/files/7911434/New.Scan.8_07.AM.Copyleaks.Report.pdf)

**Comment by himani007** [comment]
image

I have run my article on the same website provided by you and here's the results.

**Comment by Bayler** [comment]
> image
> 
> I have run my article on the same website provided by you and here's the results. 

Please fix the plagiarised texts as requested.

**Comment by himani007** [comment]
Removed plagiarism according to the report.

**Comment by Bayler** [comment]
@himani007 I'd like you to yo through all the submitted PRs to get a hang on how you can choose a good title. "Article on Hyperparameter" is not a good title.

**Comment by himani007** [comment]
Hi @Bayler, my article's title is not "Article on Hyperparameter tuning". This is the message I used to create my PR. I didn't know the message should have been the title of the article itself. 
Instead... my article's title is,
"Hyperparameter tuning in Decision Trees and Random Forests". 
Sorry if this is causing any inconvenience.

**Comment by lalith1403** [comment]
plagiarism check done ✔️

**Comment by Bayler** [comment]
@himani007 I'd like you to replace a couple of things on your links and file names> For example, in "/Hyperparmeter-Tuning/Overfitting_1.png"

1. The link and file names need to be in small letters, i.e., /hyperparmeter-tuning/overfitting_1.png
2. Separate the file name using a hyphen rather than underscore. It is good for SEO purposes, i.e., /overfitting-1.png
3. Lastly, the `jpg` pr `png` file formats need to be in small letters, i.e., jpg and not JPG

**Comment by himani007** [comment]
@Bayler, I have made the necessary changes to file names and image formats.

**Comment by Bayler** [comment]
> @Bayler, I have made the necessary changes to file names and image formats.

@himani007 You forgot about this? 

`content/authors/Himani-Gulati/avatar.JPG`

Your name and the format type all need to be in small letters

**Comment by himani007** [comment]
@Bayler, thanks. I forgot about the author file. 
Updated :heavy_check_mark:

**Comment by ahmadmardeni1** [comment]
The math equations in this tutorial are not rendered locally. Can you help with this, please? @lalith1403

**Comment by Bayler** [comment]
> The math equations in this tutorial are not rendered locally. Can you help with this, please? @lalith1403

@ahmadmardeni1 I've added a MathJax script. It should now be okay.

cc @lalith1403

**Comment by ahmadmardeni1** [comment]
Great thank you @Bayler

**Comment by himani007** [comment]
Thank you @ahmadmardeni1. 
And thank you @Bayler for your assistance :))
